src/dex/balancer-v3/balancer-v3-pool.ts

```diff
@@ -425,14 +439,24 @@ export class BalancerV3EventPool extends StatefulEventSubscriber<PoolStateMap> {
             BigInt(timestamp),
           );
         }
+        // StableSurge hook uses Amp as part of maths
+        if (step.poolState.hookType === StableSurge.type && hookState)
+          hookState = {
+            ...(hookState as StableSurgeHookState)[
+              step.poolState.poolAddress.toLowerCase()
+            ],
+            amp: step.poolState.amp,
+          };
```

Comment:
can this be relevant for non-stable pools?

Reply:
No. StableSurge is always connected to StablePool (enforced at [factory](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/pool-hooks/contracts/StableSurgePoolFactory.sol#L91))